frontend/src/components/NewRunParametersV2.tsx

```diff
@@ -257,18 +245,16 @@ function NewRunParametersV2(props: NewRunParametersProps) {
                       props.handleParameterChange(parametersInRealType);
                     }
 
-                    const { validInput, errorMessage } = checkInput(
+                    errorMessages[k] = generateInputValidationErrMsg(
                       parametersInRealType[k],
                       props.specParameters[k].parameterType,
                     );
-                    validInputs[k] = validInput;
-                    errorMessages[k] = errorMessage;
                     setErrorMessages(errorMessages);
-                    setValidInputs(validInputs);
 
-                    Object.values(validInputs).map(v2 => {
-                      allInputsValid = allInputsValid && v2;
+                    Object.values(errorMessages).map(v2 => {
```

Comment:
The key naming `v2` is a bit strange. can you change name `v2` to `errorMessage`?